core/src/CoreTelemetry.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Core
+{
+    internal class CoreTelemetry
```

Comment:
I had a look at this and have a few suggestions: 

1. the constructor of CoreTelemetry should take in an ITelemetry param. This way your class is immutable and easier to use. You should also expose properties not members, i.e. public static ITelemetry Instance { get; private set;}
2. I don't like that you have to do CoreTelemetry.Instance.GetInstance() ... Ideally we'd just have CoreTelemetry.Instance
3. The interface ITelemetry should not need to have a GetInstance()
4. It would be cleaner if the classes MSALTelemetry and ADALTelemetry would not be singletons but normal classes with state etc. and you'd use CoreTelemetry everywhere. Not sure how much work this would be though, but if it's not a lot, it's probably worth it.

Reply:
1. CoreTelemetry.instance is static and no object reference is available for coreTelemetry. I propose something like public static void InitializeCoreTelemetryService(ITelemetry instance)?
2. #Resolved
3. #Resolved
4. This looks like it can get tricky, i think I will keep it as is now for the sake of time and create a task tracking this.